### PR TITLE
Fix media query specificity

### PR DESCRIPTION
**Pull request by kof**
We haven't been sorting the media queries correctly

## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [ ] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

**Comment by istarkov** [comment]
Tested with bug we have in production, it solves it. But I think the media queries we generate look strange at least from my side.
Usually, if we are "mobile first" we don't have media query for mobile resolution

i.e. tailwind example
```
<div class="w-4 sm:w-6">ss</div>
```
generates.

```
.w-4 {
  width: 1rem;
}

@media (min-width: 640px) {
  .sm\:w-6 {
    width: 1.5rem;
  }
}
```

so mobile in the absense of `sm: lg:` applied everywhere 

We on mobile generate

```
@media all and (min-width: 360px) {
  [data-ws-id="root"] {
      flex-direction: var(--flexDirection-root, column);
      align-items: var(--alignItems-root, center);
      justify-content: var(--justifyContent-root, center);
      column-gap: var(--columnGap-root, normal);
      display: var(--display-root, flex);
  }
}
```

For me, it looks strange, what if the resolution is less than 360px?

File: packages/css-engine/src/core/css-engine.test.ts [hunk 1]
```diff
@@ -35,25 +42,57 @@ describe("CssEngine", () => {
...
-      breakpoint: mediaId,
+      breakpoint: mediaId0,
     });
     // Default media query should allways be the first to have the lowest source order specificity
     expect(engine.cssText).toMatchInlineSnapshot(`
-      "@media all {
+      "@media all and (min-width: 0px) {
+        .c1 { color: blue }
+      }
+      @media all {
```
Comment (istarkov) [inline packages/css-engine/src/core/css-engine.test.ts:55]: In queries below `@media all {` is first, comment says that default should be first but here its 2nd

File: packages/css-engine/src/core/rules.ts [hunk 2]
```diff
@@ -63,11 +63,25 @@ export type MediaRuleOptions = {
...
 export class MediaRule {
-  #options: MediaRuleOptions;
+  // Sort media rules by minWidth.
+  // Needed to ensure that more specific media rules are inserted after less specific ones.
+  // So that they get a higher specificity.
+  static sort(mediaRules: Iterable<MediaRule>) {
+    return Array.from(mediaRules).sort((a, b) => {
+      if (
+        a.options.minWidth === undefined ||
+        b.options.minWidth === undefined
```
Comment (istarkov) [inline packages/css-engine/src/core/rules.ts:73]: So it wouldn't sort but just reverse the below examples, looks like a bug as IMO both arrays after sorting should give same value
```
[{ options: { minWidth: undefined } }, { options: { minWidth: 1 } }]
// and 
[{ options: { minWidth: 1 } }, { options: { minWidth: undefined } }]
```

**Review by istarkov — CHANGES_REQUESTED** [review]

File: packages/css-engine/src/core/css-engine.test.ts (hunk 1 above)
Comment (kof) [reply packages/css-engine/src/core/css-engine.test.ts:55]: huh that's a bug, thank you

File: packages/css-engine/src/core/rules.ts (hunk 2 above)
Comment (kof) [reply packages/css-engine/src/core/rules.ts]: added a test to ensure that

**Comment by kof** [comment]
> For me, it looks strange, what if the resolution is less than 360px?

God question, I assumed its impossible, but if you are targeting a watch or something - yes, possible, our breakpoints are editable, so I think its a special case where a user might want to edit them?

I didn't set the minWidth from 0  there because the UI didn't make sense to go below  that narrow resolution.

File: packages/css-engine/src/core/rules.ts [hunk 3]
```diff
@@ -65,11 +65,22 @@ export type MediaRuleOptions = {
 };
 
 export class MediaRule {
-  #options: MediaRuleOptions;
+  // Sort media rules by minWidth.
+  // Needed to ensure that more specific media rules are inserted after less specific ones.
+  // So that they get a higher specificity.
+  static sort(mediaRules: Iterable<MediaRule>) {
+    return Array.from(mediaRules).sort((ruleA, ruleB) => {
+      return (
```
Comment (istarkov) [inline packages/css-engine/src/core/rules.ts:73]: -Infinity - (-Infinity) = NaN
Looks like undefined behavior?

Comment (kof) [reply packages/css-engine/src/core/rules.ts:73]: oh amazing behavior from javascript, will switch to -Number.MAX_SAFE_INTEGER

**Review by istarkov — APPROVED** [review]

**Pull request merged**